### PR TITLE
convert daemonset's failed pod recreation e2e test

### PR DESCRIPTION
**Pull request by crimsonfaith91**
**What this PR does / why we need it**:
This PR converts DaemonSet's failed pod recreation e2e test to an integration test.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
xref #52191

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: test/integration/daemonset/daemonset_test.go [hunk 1]
```diff
@@ -393,3 +404,92 @@ func TestInsufficientCapacityNodeDaemonDoesNotLaunchPod(t *testing.T) {
...
+		return true, nil
+	}); err != nil {
+		t.Fatalf("failed waiting for all pods to be in pending phase: %v", err)
+	}
+}
+
+// A DaemonSet should retry creating failed daemon pods
+func TestRetryCreatingFailedPod(t *testing.T) {
+	for _, strategy := range updateStrategies() {
+		server, closeFn, dc, informers, clientset := setup(t)
```
Comment (foxish) [inline test/integration/daemonset/daemonset_test.go]: Should this setup be done once per updateStrategy? I think it should be once per test looking at [other examples](https://github.com/kubernetes/kubernetes/blob/master/test/integration/replicaset/replicaset_test.go).

File: test/integration/daemonset/daemonset_test.go [hunk 2]
```diff
@@ -393,3 +404,92 @@ func TestInsufficientCapacityNodeDaemonDoesNotLaunchPod(t *testing.T) {
...
+		validateDaemonSetStatus(dsClient, ds.Name, ds.Namespace, 1, t)
+
+		// Set the pod's phase to "Failed"
+		pods := getPods(t, podClient, testLabels())
+		updatePodStatus(t, podClient, pods.Items[0].Name, func(pod *v1.Pod) {
+			pod.Status.Phase = v1.PodFailed
+		})
+
+		// Wait for the failed pod to be created again and in pending phase
+		waitForAllPodsPending(podClient, t)
```
Comment (foxish) [inline test/integration/daemonset/daemonset_test.go]: Why stop when it's pending? Doesn't it go to running?

File: test/integration/daemonset/daemonset_test.go [hunk 3]
```diff
@@ -393,3 +404,92 @@ func TestInsufficientCapacityNodeDaemonDoesNotLaunchPod(t *testing.T) {
...
+		if err != nil {
+			t.Fatalf("failed to create node %q: %v", nodeName, err)
+		}
+
+		// Validate everything works correctly, include marking pods
+		// as ready and ensure they launch on every node
+		validateDaemonSetPodsAndMarkReady(podClient, podInformer, 1, t)
+		validateDaemonSetStatus(dsClient, ds.Name, ds.Namespace, 1, t)
+
+		// Set the pod's phase to "Failed"
```
Comment (foxish) [inline test/integration/daemonset/daemonset_test.go]: Can we test if the pod is bound to the node that was created above?

**Review by foxish — COMMENTED** [review]
Few comments - also, does the e2e have value by itself - say, for conformance?

**Comment by foxish** [comment]
Based on the discussion, we probably don't need to remove this e2e.

cc/ @enisoc

File: test/integration/daemonset/daemonset_test.go (hunk 1 above)
Comment (crimsonfaith91) [reply test/integration/daemonset/daemonset_test.go]: For other daemonset integration tests, the setup is done once per updateStrategy.

File: test/integration/daemonset/daemonset_test.go (hunk 2 above)
Comment (crimsonfaith91) [reply test/integration/daemonset/daemonset_test.go]: The pods will not get ready if we do not explicitly mark it, so they will stay at pending phase. Am I right?

File: test/integration/daemonset/daemonset_test.go (hunk 3 above)
Comment (crimsonfaith91) [reply test/integration/daemonset/daemonset_test.go]: sure, i will add the check.

**Comment by crimsonfaith91** [comment]
As sync by person, we are not sure with purpose of the e2e test, but we are confident that the e2e test will be part of the conformance. Hence, I will open an issue investigating purpose of the e2e test and close this PR.

**Pull request closed without merging**